src/client/pps/pps.proto

```diff
@@ -110,6 +106,28 @@ message ParallelismSpec {
   double coefficient = 3;
 }
 
+message ObjectStore {
+  string URL = 1;
+}
+
+message SQLDb {
+  string driver = 1;
+  string URL = 2;
+  repeated string init = 3;
+  repeated string tables = 4;
+}
+
+enum ConnectorType {
+  OBJECT_STORE = 0;
+  SQL_DB = 1;
+}
+
+message Connector {
+    ConnectorType type = 1;
```

Comment:
Yeah I think that makes sense, a few questions:

- What do you think is the right behavior if multiple keys are set? The two options I see are error or use the "first" value where first would just be defined by which order we check them in? I think the latter might be simpler and is unlikely to cause many problems in practice since I don't think people will be having duplicate entries and if they do at least this will be deterministic behavior so hopefully not too confusing.

- Any reason that I shouldn't migrate the ParallelismSpec code over to this method as well? Seems like it shouldn't be too hard and is basically backward compatible with existing pipelines.